client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDatabaseMetaData.java

```diff
@@ -1357,6 +1389,102 @@ public void testGetColumnsMetadataCalls()
                         .withGetColumnsCount(3000));
     }
 
+    @Test
+    public void testAssumeLiteralMetadataCalls()
+            throws Exception
+    {
+        try (Connection connection = DriverManager.getConnection(
+                format("jdbc:trino://%s?assumeLiteralNamesInMetadataCallsForNonConformingClients=true", server.getAddress()),
+                "admin",
+                null)) {
+            // getTables's schema name pattern treated as literal
+            assertMetadataCalls(
+                    connection,
+                    readMetaData(
+                            databaseMetaData -> databaseMetaData.getTables(COUNTING_CATALOG, "test_schema1", null, null),
```

Comment:
would this return something else if `assumeLiteralNamesInMetadataCallsForNonConformingClients` was not passed?
Or in other words. Is there some other schema which matches `test_schema1` (e.g `testXschema1`)

Reply:
no, but it would result in a different number of metadata calls.

note however the `test_schema_` test case which seems what you're looking for.

Reply:
make sense